C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift

```diff
@@ -53,14 +53,18 @@ final class URLSessionApiDataSource: ApiDataSource {
                 completion(.failure(apiError))
                 logger.log(apiError)
             } else if let data = data {
-                do {
-                    let sanitizedData = data.isEmpty ? "{}".data(using: .utf8)! : data
-                    let object = try jsonDecoder.decode(T.self, from: sanitizedData)
-                    completion(.success(object))
-                } catch let error {
-                    let apiError = ApiError.cannotParseData(error.localizedDescription)
-                    logger.log(apiError)
-                    completion(.failure(apiError))
+                if T.self == Data.self {
+                    completion(.success(data as! T))
```

Comment:
Do you need this force casting here? What is the type of data returned by sender?

Reply:
Yes, this function doesn't know what the type `T` is, so I need to force cast it here.